### PR TITLE
feat: datacoord/node watch based on rpc

### PR DESCRIPTION
**Pull request by yiwangdr**
issue: https://github.com/milvus-io/milvus/issues/25309

### CODE REVIEW

**Comment by xiaofan-luan** [comment]
@yiwangdr 
let's merge this ASAP and let the QA team test on it.

**Comment by xiaofan-luan** [comment]
@yanliang567 
the e2e test seems to be really unstable these days and we get many random failings

**Comment by yanliang567** [comment]
we have had a few server running for ci/e2e exclusively, please retry if it fails for env. issues.

**Comment by yiwangdr** [comment]
rerun ut

**Comment by yiwangdr** [comment]
rerun ut

File: internal/datacoord/channel.go [hunk 1]
```diff
@@ -39,7 +44,30 @@ type RWChannel interface {
...
 
-var _ RWChannel = (*channelMeta)(nil)
+func NewRWChannel(name string,
+	collectionID int64,
+	startPos []*commonpb.KeyDataPair,
+	schema *schemapb.CollectionSchema,
+	createTs uint64,
+) RWChannel {
+	if paramtable.Get().DataCoordCfg.EnableBalanceChannelWithRPC.GetAsBool() {
+		return &StateChannel{
```
Comment (XuanYang-cn) [inline internal/datacoord/channel.go:54]: This StateChannel looks invalid for missing currentState and assignedNode, recommand to set currentState="standby" and assignedNode=bufferID

**Review by XuanYang-cn — APPROVED** [review]
/lgtm

**Pull request merged**